include/aws/common/byte_buf.h

```diff
@@ -108,7 +115,10 @@ bool aws_byte_buf_eq(const struct aws_byte_buf *a, const struct aws_byte_buf *b)
  * long enough to use the results.
  */
 AWS_COMMON_API
-int aws_byte_buf_split_on_char(struct aws_byte_buf *input_str, char split_on, struct aws_array_list *output);
+int aws_byte_cursor_split_on_char(
```

Comment:
Where is the cursor left at the end? If e.g. the output array_list has a limited size I could see it being useful to have the cursor be at the point where it left off.

Reply:
Good thinking, I'll make it do that.

Reply:
Just kidding, that only applies to the N permutation. This version just keeps appending to the list for ever (assumes dynamic list).